subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -49,27 +49,21 @@ This is a potentially breaking change if you are consuming the console output of
 
 === Deprecations
 
-[[custom_configuration_roles]]
-==== Custom configuration roles
-
-// TODO: There is currently no API for users to provide "custom roles" or even
-// any roles at all other than the existing role-mutating methods. We should revisit
-// this deprecation log and section when we introduce a public API for creating
-// locked-role configurations.
-
-Custom roles have been deprecated.
-Use a pre-defined role instead.
-
 [[configurations_allowed_usage]]
 ==== Allowed configurations usage
 
-The usage of configurations should be fixed at creation.
-Mutating the allowed usage on a configuration is deprecated.
+The allowed usage of a configuration should be immutable after creation.
+Mutating the allowed usage on a configuration created by a Gradle core plugin is deprecated.
 This includes calling any of the following `Configuration` methods:
 
 - `setCanBeConsumed(boolean)`
 - `setCanBeResolved(boolean)`
 
+This will now emit a deprecation warning, except for certain special cases which deal with legacy configurations or make allowances for the existing behavior of popular plugins.
```

Comment:
```suggestion
This will now emit a deprecation warning, except for certain special cases which deal with legacy configurations.
```

I'm not sure we need to mention that we're allowing Kotlin to do some special things. I think the general message should be "this isn't gonna be allowed anymore". If the message says we've special cased some things, some users might get the idea to abuse those special cases.

Reply:
I want to leave this caveat in.  We don't mention Kotlin by name, but I want to give a reason why we're not doing this everywhere, that we can point to if someone files a bug saying "this didn't warn me like it should have".

Reply:
In that case, I would prefer to not use wording like `make allowances`. What do you think about something like this: 
```suggestion
Calling these methods on core configurations will now emit a deprecation warning. 
This rule does not yet apply to detached configurations or configurations created 
in buildscripts and third-party plugins. 
As a special case, calling `setCanBeConsumed(false)` on `apiElements` or `runtimeElements`
is not yet deprecated in order to avoid warnings that would be otherwise emitted when
using select popular third-party plugins. 
```

Reply:
I'm confused by your response here. My suggestion is very explicit with what the special cases are

Reply:
My mistake, reading too fast and only read the highlighted part of the change.  How about now.